### PR TITLE
Preserve raw tokens for strings and regexps

### PR DESCRIPTION
**Pull request by dennwc**
Currently, JS driver uses unescaped string and regexp literals from the native AST as `@token`. This leads to confusion for ML, since they expect string token to cover the whole node in the source code (according to the positional information).

Although in general, this will require adding a separate "token stream" to the UAST, those two issues can be solved in the current version.

See comments in related issues for details (link is in the commit message).

### CODE REVIEW

File: driver/normalizer/normalizer.go [hunk 1]
```diff
@@ -243,3 +285,141 @@ var Normalizers = []Mapping{
...
+	if !ok {
+		return nil, ErrUnexpectedType.New(nodes.String(""), n)
+	}
+	s := string(sn)
+	s = quoteSingle(s)
+	return nodes.String(s), nil
+}
+
+func unquoteSingle(s string) (string, error) {
+	// copied from strconv.Unquote
```
Comment (creachadair) [inline driver/normalizer/normalizer.go]: Why copied?

Comment (dennwc) [reply driver/normalizer/normalizer.go]: It needs to handle unescaping of `'` quotes.

Comment (creachadair) [reply driver/normalizer/normalizer.go]: Can you please add a comment to that effect somewhere nearby?

Comment (dennwc) [reply driver/normalizer/normalizer.go]: Done!

**Comment by bzz** [comment]
👍 

If I understood the change,  it boils down to keeping whole literals in `@token`:

Code: string literal `attribute="If parent is JSX keep double quote"`
Before
 - Node @token `If parent is JSX keep double quote`
After
 - Node @token `"If parent is JSX keep double quote"`

and same for 
Code:  regexp literal `/a/g`
Before
 - Node @token `a`
After
 - Node @token `/a/g`

@dennwc could you please help me understand, how/if we want it to be consistent with behavior of other drivers?

**Comment by dennwc** [comment]
@bzz Other drivers already include quotes in `@token` for Native and Annotated mode. This is only an issue with the JS driver that uses wrong fields from native AST.

As for `strconv.go`, I don't think it requires extensive test coverage since it's copied from the Go stdlib, which already ensures that the code is correct. But we can add a few small test cases to detect regressions.

**Comment by bzz** [comment]
On

>>See comments in related issues for details.
>
>not sure which issue is that.

Have missed those in commit messages -  #22  and #32

File: driver/normalizer/normalizer.go [hunk 2]
```diff
@@ -243,3 +283,42 @@ var Normalizers = []Mapping{
...
+func (op singleQuote) Construct(st *State, n nodes.Node) (nodes.Node, error) {
+	n, err := op.op.Construct(st, n)
+	if err != nil {
+		return nil, err
+	}
+	sn, ok := n.(nodes.String)
+	if !ok {
+		return nil, ErrUnexpectedType.New(nodes.String(""), n)
+	}
+	s := string(sn)
```
Comment (creachadair) [inline driver/normalizer/normalizer.go]: `return nodes.String(quoteSingle(string(sn))), nil`

Or at least `s := quoteSingle(string(sn))`, reassignment is really not needed here and makes my whiskers twitch. :)

**Pull request merged**